### PR TITLE
Dice

### PR DESCRIPTION
**Pull request by grxver7**
Adding Dice Scene to main. 

#CanvasLayer
#Centered
#No animations, causing issues

### CODE REVIEW

**Review by ubeF — CHANGES_REQUESTED** [review]
Removed

**Review by ubeF — DISMISSED** [review]

File: dice/Dice.gd [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 func _on_button_pressed():
-	var random = randi_range(1,6)
+	var random = randi_range(1,12)
 	
-	animation.play("popUpScale")
 	label.set_text(String("%d" % random))
-	animation.play("popUpScale")
 	button.hide()
+	await get_tree().create_timer(1.5).timeout
+	self.visible = false
```
Comment (hfxbse) [inline dice/Dice.gd]: No need to prefix `visible` with `self`.

**Review by hfxbse — CHANGES_REQUESTED** [review]
Looks good so far. Something that is left to do is to add a coroutine that shows the dialog and returns the dice result. The idea is to [call and await the coroutine in the main loop](https://github.com/hfxbse/mparty/blob/a35bc6fcc572574d37810c52c1401e03263e8422/main/main.gd#L84).

File: dice/Dice.gd (hunk 1 above)
Comment (KevinBasener) [reply dice/Dice.gd]: Connected the dice menu to the player movement in main.

File: menu_overlay/menu_overlay.gd [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+var dice_menu
+var difficulty_menu
+	
+func add_dice_menu():
+	dice_menu = preload("res://dice/Dice.tscn").instantiate()
+	add_child(dice_menu)
+	
+	var dice_value = await dice_menu.dice_number
+	await get_tree().create_timer(1.5).timeout
```
Comment (hfxbse) [inline menu_overlay/menu_overlay.gd]: Kinda a weird feeling having an additional delay after the dice interface disappears.

**Review by hfxbse — COMMENTED** [review]
Not really a fan of adding a scene to instantiate the dice scene, which seems overly complicated. The wording `add_dice_menu` also does give any indication that it is performing the dice roll at all. Why not instantiate the scene directly and await the signal afterward, then clean up?

The files within `dice` should also be renamed to start with a lowercase letter.

**Comment by KevinBasener** [comment]
> Not really a fan of adding a scene to instantiate the dice scene, which seems overly complicated. The wording `add_dice_menu` also does give any indication that it is performing the dice roll at all. Why not instantiate the scene directly and await the signal afterward, then clean up?
> 
> The files within `dice` should also be renamed to start with a lowercase letter.

The idea was to have an extra layer other than hud, to display all the dialogs that might be added. Therefore keeping the hud elements and dialogs on separate layers. Should I instead add the dice to the hud or even directly to the main tree?

**Comment by hfxbse** [comment]
> > Not really a fan of adding a scene to instantiate the dice scene, which seems overly complicated. The wording `add_dice_menu` also does give any indication that it is performing the dice roll at all. Why not instantiate the scene directly and await the signal afterward, then clean up?
> > The files within `dice` should also be renamed to start with a lowercase letter.
> 
> The idea was to have an extra layer other than hud, to display all the dialogs that might be added. Therefore keeping the hud elements and dialogs on separate layers. Should I instead add the dice to the hud or even directly to the main tree?

Do not put it in the HUD, it is a dialog and has nothing to do with it. Just keep it as simple as possible and insert it directly into the main tree. I do not see how an extra layer would add any real value, as that layer needs to be added a different sub-trees if implemented, but not every sub-tree needs for example the dice dialog.

**Comment by KevinBasener** [comment]
> > > Not really a fan of adding a scene to instantiate the dice scene, which seems overly complicated. The wording `add_dice_menu` also does give any indication that it is performing the dice roll at all. Why not instantiate the scene directly and await the signal afterward, then clean up?
> > > The files within `dice` should also be renamed to start with a lowercase letter.
> > 
> > 
> > The idea was to have an extra layer other than hud, to display all the dialogs that might be added. Therefore keeping the hud elements and dialogs on separate layers. Should I instead add the dice to the hud or even directly to the main tree?
> 
> Do not put it in the HUD, it is a dialog and has nothing to do with it. Just keep it as simple as possible and insert it directly into the main tree. I do not see how an extra layer would add any real value, as that layer needs to be added a different sub-trees if implemented, but not every sub-tree needs for example the dice dialog.

Putting it directly into the main tree, leads to the problem, that the dice dialog behaves like the map when the main camera is used. It will stay at the same coordinates in the scene (like map), when the main camera is drag clicked, but it will not be at a fixed place (e.g. center) of the window.

**Comment by hfxbse** [comment]
> Putting it directly into the main tree, leads to the problem, that the dice dialog behaves like the map when the main camera is used. It will stay at the same coordinates in the scene (like map), when the main camera is drag clicked, but it will not be at a fixed place (e.g. center) of the window.

Can this not be addressed by changing the root node of the dice panel scene to a canvas like it has been done in other scenes before?

**Comment by KevinBasener** [comment]
> > Putting it directly into the main tree, leads to the problem, that the dice dialog behaves like the map when the main camera is used. It will stay at the same coordinates in the scene (like map), when the main camera is drag clicked, but it will not be at a fixed place (e.g. center) of the window.
> 
> Can this not be addressed by changing the root node of the dice panel scene to a canvas like it has been done in other scenes before?

Oh yes, true. Now its fixed.

**Review by hfxbse — APPROVED** [review]

**Pull request merged**